src/actions/UserAction.ts

```diff
@@ -0,0 +1,39 @@
+import { userConstants } from '../config';
+import * as userService from '../services/User';
+
+export enum UserActionType {
+  LOGIN = 'login',
+  LOGOUT = 'logout'
+}
+
+export const request = user => {
+  return { type: userConstants.LOGIN_REQUEST, user };
+};
+
+export const success = user => {
+  return { type: userConstants.LOGIN_SUCCESS, user };
+};
+
+export const failure = error => {
+  return { type: userConstants.LOGIN_FAILURE, error };
+};
+
+export const login = (username: string, password: string) => {
+  return dispatch => {
```

Comment:
I'm mostly learning with this review, as I haven't worked with Redux yet. So, this "dispatch" function here is the convention of redux for actions. Am I right ?

Reply:
Yes, you can see an example here https://egghead.io/lessons/react-redux-store-methods-getstate-dispatch-and-subscribe about getstate dispatch and subscribe